crates/db/src/traits.rs

```diff
@@ -226,6 +231,8 @@ pub trait ChainstateDatabase {
 
     /// Gets the toplevel chain state at a particular block index (height).
     fn get_toplevel_state(&self, idx: u64) -> DbResult<Option<Chainstate>>;
+
+    fn get_state_at(&self, idx: u64) -> DbResult<Option<Chainstate>>;
```

Comment:
Same thing as before, don't make the database reconstruct this.  We can reconstruct it ourselves.  And right now we store all states anyways, so you can just call `get_toplevel_state`.

Reply:
Doing it in `rpc_server.rs` is fine? Please comment on the next commit that I'm gonna make.